### PR TITLE
Make the AWS session token duration a configurable setting

### PR DESCRIPTION
**Pull request by onkarbhat**
## Change Overview

- Make the AWS session token duration a configurable setting

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/secrets/aws_test.go [hunk 1]
```diff
@@ -75,9 +76,9 @@ func (s *AWSSecretSuite) TestExtractAWSCredentials(c *C) {
 			errChecker: NotNil,
 		},
 	}
-	for _, tc := range tcs {
-		creds, err := ExtractAWSCredentials(context.Background(), tc.secret)
-		c.Check(creds, DeepEquals, tc.expected)
+	for testNum, tc := range tcs {
+		creds, err := ExtractAWSCredentials(context.Background(), tc.secret, aws.AssumeRoleDurationDefault)
+		c.Check(creds, DeepEquals, tc.expected, Commentf("test number:%d", testNum))
```
Comment (pavannd1) [inline pkg/secrets/aws_test.go:81]:
```suggestion
		c.Check(creds, DeepEquals, tc.expected, Commentf("test number: %d", testNum))
```

**Review by pavannd1 — APPROVED** [review]
LGTM

File: pkg/aws/aws.go [hunk 2]
```diff
@@ -66,8 +82,10 @@ func GetCredentials(ctx context.Context, config map[string]string) (*credentials
 		if err != nil {
 			return nil, errors.Wrap(err, "Failed to create session to initialize Web Identify credentials")
 		}
-		// If we have credentials to use with a Web Identity provider - use those
-		creds = stscreds.NewWebIdentityCredentials(sess, os.Getenv(roleARNEnvKey), "", os.Getenv(webIdentityTokenFilePathEnvKey))
+		svc := sts.New(sess)
+		p := stscreds.NewWebIdentityRoleProvider(svc, os.Getenv(roleARNEnvKey), "", os.Getenv(webIdentityTokenFilePathEnvKey))
+		p.Duration = assumeRoleDuration
```
Comment (pavannd1) [inline pkg/aws/aws.go:87]: Should we add a comment here to note why we need to update this separately?

File: pkg/secrets/aws.go [hunk 3]
```diff
@@ -64,14 +67,24 @@ func ValidateAWSCredentials(secret *v1.Secret) error {
 //
 // If the type of the secret is not "secret.kanister.io/aws", it returns an error.
 // If the required types are not avaialable in the secrets, it returns an errror.
-func ExtractAWSCredentials(ctx context.Context, secret *v1.Secret) (*credentials.Value, error) {
+//
+// ExtractAWSCredentials accepts an assumeRoleDuration which is used to set
+// the duration of the AWS session token provided by K10.
```
Comment (pavannd1) [inline pkg/secrets/aws.go]:
```suggestion
// ExtractAWSCredentials accepts an assumeRoleDuration which is used to set
// the duration of the AWS session token provided.
```

File: pkg/aws/aws.go (hunk 2 above)
Comment (onkarbhat) [reply pkg/aws/aws.go]: Good idea. I've moved the code for getting creds with a duration to a helper function and added comments for that function.

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**